// vue-toaster/toaster-component.js
;(function(){
module.exports = {
  mixins: [require("vue-mixins/getViewportSize"), require("vue-mixins/vue")],
  components: {
    toast: require("./toast")
  },
  props: {
    id: {
      type: String,
      "default": "toast-container"
    },
    isTop: {
      "default": null
    },
    component: {
      type: String,
      "default": "toast"
    },
    "class": {
      type: Array,
      "default": function() {
        return ["toaster"];
      }
    },
    toastClass: {
      type: Array,
      "default": function() {
        return ["toast"];
      }
    },
    zIndex: {
      type: Number,
      "default": 10000
    },
    timeout: {
      type: Number,
      "default": 2500
    },
    transition: {
      "default": "toaster-transition"
    }
  },
  computed: {
    cClass: function() {
      return this["class"];
    },
    cTransition: function() {
      var comp, name;
      name = this.transition;
      this.transitionName = "toast";
      comp = this.Vue.util.resolveAsset(this.$options, 'components', name);
      if (comp != null) {
        this.$options.components[name] = comp;
        this.transitionName = null;
      } else {
        name = "transition-group";
      }
      return name;
    }
  },
  data: function() {
    return {
      toasts: [],
      transitionName: "toast",
      used: 0
    };
  },
  mounted: function() {
    return this.$nextTick(function() {
      var pos, vpheight;
      if (this.isTop == null) {
        pos = this.$el.getBoundingClientRect();
        vpheight = this.getViewportSize().height;
        return this.isTop = pos.top + pos.height / 2 <= vpheight / 2;
      }
    });
  },
  methods: {
    clear: function() {
      var i, len, ref, toast;
      ref = this.toasts;
      for (i = 0, len = ref.length; i < len; i++) {
        toast = ref[i];
        toast.transition = null;
      }
      return this.toasts = [];
    },
    toast: function(options) {
      if (options == null) {
        options = {};
      }
      if (options.component == null) {
        options.component = this.component;
      }
      if (options.timeout == null) {
        options.timeout = this.timeout;
      }
      if (options["class"] == null) {
        options["class"] = this.toastClass;
      }
      options.close = (function(_this) {
        return function(e) {
          var index;
          if (e != null) {
            if (e.defaultPrevented) {
              return;
            }
            e.preventDefault();
          }
          index = _this.toasts.indexOf(options);
          if (index > -1) {
            _this.toasts.splice(index, 1);
            if (typeof options.removeTimeout === "function") {
              options.removeTimeout();
            }
            return typeof options.cb === "function" ? options.cb() : void 0;
          }
        };
      })(this);
      options.setTimeout = function() {
        var timeoutObj;
        if (typeof options.removeTimeout === "function") {
          options.removeTimeout();
        }
        if (options.timeout) {
          timeoutObj = setTimeout(options.close, options.timeout);
        }
        return options.removeTimeout = function() {
          if (timeoutObj != null) {
            return clearTimeout(timeoutObj);
          }
        };
      };
      options.setTimeout();
      if (this.isTop) {
        this.toasts.push(options);
      } else {
        this.toasts.unshift(options);
      }
      return options;
    }
  }
};

})()
if (module.exports.__esModule) module.exports = module.exports.default
var __vue__options__ = (typeof module.exports === "function"? module.exports.options: module.exports)
__vue__options__.render = function(){with(this){return _h(cTransition,{tag:"transition-group",class:cClass,style:({position:'fixed',zIndex:zIndex}),attrs:{"tag":"div","id":id,"name":transitionName}},[_l((toasts),function(toast){return _h(toast.component,{key:toast,tag:"component",class:toast.class,style:(toast.style),attrs:{"options":toast},on:{"close":toast.close},nativeOn:{"click":function($event){toast.close($event)},"mouseenter":function($event){toast.removeTimeout($event)},"mouseleave":function($event){toast.setTimeout($event)}}})})])}}
__vue__options__.staticRenderFns = []
