// vue-form-kit/dist/TextArea.be0a0328.js
import { openBlock as d, createBlock as i, resolveDynamicComponent as o, createSlots as m, withCtx as r, createElementVNode as f, renderList as s, renderSlot as g, normalizeProps as c, guardReactiveProps as y } from "vue";
import { _ as S } from "./_plugin-vue_export-helper.9d74fd37.js";
const p = {
  name: "FormTextArea",
  props: {
    modelValue: {
      type: String,
      default: ""
    },
    tabindex: {
      type: String,
      default: null
    },
    label: {
      type: String,
      default: null
    },
    placeholder: {
      type: String,
      default: null
    },
    error: {
      type: String,
      default: null
    },
    id: {
      type: String,
      default: null
    },
    name: {
      type: String,
      default: null
    },
    autocomplete: {
      type: String,
      default: null
    },
    maxlength: {
      type: String,
      default: null
    },
    minlength: {
      type: String,
      default: null
    },
    required: {
      type: [String, Boolean],
      default: !1
    },
    disabled: {
      type: [String, Boolean],
      default: !1
    },
    cols: {
      type: [Number, String],
      default: null
    },
    rows: {
      type: [Number, String],
      default: null
    },
    dusk: {
      type: String,
      default: null
    },
    inputWrapClass: {
      type: [String, Object, Array],
      default: null
    }
  },
  emits: ["update:modelValue"]
}, b = ["id", "value", "dusk", "tabindex", "name", "autocomplete", "required", "disabled", "placeholder", "maxlength", "minlength", "cols", "rows"];
function x(t, l, e, h, w, q) {
  return d(), i(o(`${t.$formKitConfigs.componentPrefix}FormInputWrapper`), {
    id: e.id,
    label: e.label,
    error: e.error,
    required: e.required,
    "input-wrap-class": e.inputWrapClass
  }, m({
    default: r(() => [
      f("textarea", {
        id: e.id,
        value: e.modelValue,
        dusk: e.dusk || `form--${e.name || e.id || "input"}`,
        tabindex: e.tabindex,
        name: e.name,
        autocomplete: e.autocomplete,
        required: e.required,
        disabled: e.disabled,
        class: "form-input-text form-input-textarea w-full",
        placeholder: e.placeholder,
        maxlength: e.maxlength,
        minlength: e.minlength,
        cols: e.cols,
        rows: e.rows,
        onInput: l[0] || (l[0] = (a) => t.$emit("update:modelValue", a.target.value))
      }, null, 40, b)
    ]),
    _: 2
  }, [
    s(t.$slots, (a, n) => ({
      name: n,
      fn: r((u) => [
        g(t.$slots, n, c(y(u)))
      ])
    }))
  ]), 1032, ["id", "label", "error", "required", "input-wrap-class"]);
}
const v = /* @__PURE__ */ S(p, [["render", x]]);
export {
  v as default
};
