// vue-form-kit/dist/vue-form-kit.js
import { defineAsyncComponent as J } from "vue";
var i = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function K(o, e, r) {
  var t = -1, n = o.length;
  e < 0 && (e = -e > n ? 0 : n + e), r = r > n ? n : r, r < 0 && (r += n), n = e > r ? 0 : r - e >>> 0, e >>>= 0;
  for (var u = Array(n); ++t < n; )
    u[t] = o[t + e];
  return u;
}
var Y = K, Q = Y;
function X(o, e, r) {
  var t = o.length;
  return r = r === void 0 ? t : r, !e && r >= t ? o : Q(o, e, r);
}
var oo = X, eo = "\\ud800-\\udfff", ro = "\\u0300-\\u036f", no = "\\ufe20-\\ufe2f", to = "\\u20d0-\\u20ff", uo = ro + no + to, co = "\\ufe0e\\ufe0f", ao = "\\u200d", io = RegExp("[" + ao + eo + uo + co + "]");
function so(o) {
  return io.test(o);
}
var h = so;
function po(o) {
  return o.split("");
}
var mo = po, y = "\\ud800-\\udfff", fo = "\\u0300-\\u036f", vo = "\\ufe20-\\ufe2f", lo = "\\u20d0-\\u20ff", bo = fo + vo + lo, go = "\\ufe0e\\ufe0f", xo = "[" + y + "]", p = "[" + bo + "]", m = "\\ud83c[\\udffb-\\udfff]", Io = "(?:" + p + "|" + m + ")", R = "[^" + y + "]", T = "(?:\\ud83c[\\udde6-\\uddff]){2}", A = "[\\ud800-\\udbff][\\udc00-\\udfff]", $o = "\\u200d", O = Io + "?", j = "[" + go + "]?", So = "(?:" + $o + "(?:" + [R, T, A].join("|") + ")" + j + O + ")*", Co = j + O + So, ho = "(?:" + [R + p + "?", p, T, A, xo].join("|") + ")", yo = RegExp(m + "(?=" + m + ")|" + ho + Co, "g");
function Ro(o) {
  return o.match(yo) || [];
}
var To = Ro, Ao = mo, Oo = h, jo = To;
function Uo(o) {
  return Oo(o) ? jo(o) : Ao(o);
}
var ko = Uo, wo = typeof i == "object" && i && i.Object === Object && i, _o = wo, Lo = _o, Fo = typeof self == "object" && self && self.Object === Object && self, Eo = Lo || Fo || Function("return this")(), Mo = Eo, Po = Mo, Bo = Po.Symbol, f = Bo;
function Wo(o, e) {
  for (var r = -1, t = o == null ? 0 : o.length, n = Array(t); ++r < t; )
    n[r] = e(o[r], r, o);
  return n;
}
var zo = Wo, Do = Array.isArray, No = Do, d = f, U = Object.prototype, Zo = U.hasOwnProperty, Go = U.toString, a = d ? d.toStringTag : void 0;
function Ho(o) {
  var e = Zo.call(o, a), r = o[a];
  try {
    o[a] = void 0;
    var t = !0;
  } catch {
  }
  var n = Go.call(o);
  return t && (e ? o[a] = r : delete o[a]), n;
}
var Vo = Ho, qo = Object.prototype, Jo = qo.toString;
function Ko(o) {
  return Jo.call(o);
}
var Yo = Ko, v = f, Qo = Vo, Xo = Yo, oe = "[object Null]", ee = "[object Undefined]", l = v ? v.toStringTag : void 0;
function re(o) {
  return o == null ? o === void 0 ? ee : oe : l && l in Object(o) ? Qo(o) : Xo(o);
}
var ne = re;
function te(o) {
  return o != null && typeof o == "object";
}
var ue = te, ce = ne, ae = ue, ie = "[object Symbol]";
function se(o) {
  return typeof o == "symbol" || ae(o) && ce(o) == ie;
}
var pe = se, b = f, me = zo, fe = No, de = pe, ve = 1 / 0, g = b ? b.prototype : void 0, x = g ? g.toString : void 0;
function k(o) {
  if (typeof o == "string")
    return o;
  if (fe(o))
    return me(o, k) + "";
  if (de(o))
    return x ? x.call(o) : "";
  var e = o + "";
  return e == "0" && 1 / o == -ve ? "-0" : e;
}
var le = k, be = le;
function ge(o) {
  return o == null ? "" : be(o);
}
var s = ge, xe = oo, Ie = h, $e = ko, Se = s;
function Ce(o) {
  return function(e) {
    e = Se(e);
    var r = Ie(e) ? $e(e) : void 0, t = r ? r[0] : e.charAt(0), n = r ? xe(r, 1).join("") : e.slice(1);
    return t[o]() + n;
  };
}
var he = Ce, ye = he, Re = ye("toUpperCase"), w = Re, Te = s, Ae = w;
function Oe(o) {
  return Ae(Te(o).toLowerCase());
}
var je = Oe;
function Ue(o, e, r, t) {
  var n = -1, u = o == null ? 0 : o.length;
  for (t && u && (r = o[++n]); ++n < u; )
    r = e(r, o[n], n, o);
  return r;
}
var ke = Ue;
function we(o) {
  return function(e) {
    return o == null ? void 0 : o[e];
  };
}
var _e = we, Le = _e, Fe = {
  \u00C0: "A",
  \u00C1: "A",
  \u00C2: "A",
  \u00C3: "A",
  \u00C4: "A",
  \u00C5: "A",
  \u00E0: "a",
  \u00E1: "a",
  \u00E2: "a",
  \u00E3: "a",
  \u00E4: "a",
  \u00E5: "a",
  \u00C7: "C",
  \u00E7: "c",
  \u00D0: "D",
  \u00F0: "d",
  \u00C8: "E",
  \u00C9: "E",
  \u00CA: "E",
  \u00CB: "E",
  \u00E8: "e",
  \u00E9: "e",
  \u00EA: "e",
  \u00EB: "e",
  \u00CC: "I",
  \u00CD: "I",
  \u00CE: "I",
  \u00CF: "I",
  \u00EC: "i",
  \u00ED: "i",
  \u00EE: "i",
  \u00EF: "i",
  \u00D1: "N",
  \u00F1: "n",
  \u00D2: "O",
  \u00D3: "O",
  \u00D4: "O",
  \u00D5: "O",
  \u00D6: "O",
  \u00D8: "O",
  \u00F2: "o",
  \u00F3: "o",
  \u00F4: "o",
  \u00F5: "o",
  \u00F6: "o",
  \u00F8: "o",
  \u00D9: "U",
  \u00DA: "U",
  \u00DB: "U",
  \u00DC: "U",
  \u00F9: "u",
  \u00FA: "u",
  \u00FB: "u",
  \u00FC: "u",
  \u00DD: "Y",
  \u00FD: "y",
  \u00FF: "y",
  \u00C6: "Ae",
  \u00E6: "ae",
  \u00DE: "Th",
  \u00FE: "th",
  \u00DF: "ss",
  \u0100: "A",
  \u0102: "A",
  \u0104: "A",
  \u0101: "a",
  \u0103: "a",
  \u0105: "a",
  \u0106: "C",
  \u0108: "C",
  \u010A: "C",
  \u010C: "C",
  \u0107: "c",
  \u0109: "c",
  \u010B: "c",
  \u010D: "c",
  \u010E: "D",
  \u0110: "D",
  \u010F: "d",
  \u0111: "d",
  \u0112: "E",
  \u0114: "E",
  \u0116: "E",
  \u0118: "E",
  \u011A: "E",
  \u0113: "e",
  \u0115: "e",
  \u0117: "e",
  \u0119: "e",
  \u011B: "e",
  \u011C: "G",
  \u011E: "G",
  \u0120: "G",
  \u0122: "G",
  \u011D: "g",
  \u011F: "g",
  \u0121: "g",
  \u0123: "g",
  \u0124: "H",
  \u0126: "H",
  \u0125: "h",
  \u0127: "h",
  \u0128: "I",
  \u012A: "I",
  \u012C: "I",
  \u012E: "I",
  \u0130: "I",
  \u0129: "i",
  \u012B: "i",
  \u012D: "i",
  \u012F: "i",
  \u0131: "i",
  \u0134: "J",
  \u0135: "j",
  \u0136: "K",
  \u0137: "k",
  \u0138: "k",
  \u0139: "L",
  \u013B: "L",
  \u013D: "L",
  \u013F: "L",
  \u0141: "L",
  \u013A: "l",
  \u013C: "l",
  \u013E: "l",
  \u0140: "l",
  \u0142: "l",
  \u0143: "N",
  \u0145: "N",
  \u0147: "N",
  \u014A: "N",
  \u0144: "n",
  \u0146: "n",
  \u0148: "n",
  \u014B: "n",
  \u014C: "O",
  \u014E: "O",
  \u0150: "O",
  \u014D: "o",
  \u014F: "o",
  \u0151: "o",
  \u0154: "R",
  \u0156: "R",
  \u0158: "R",
  \u0155: "r",
  \u0157: "r",
  \u0159: "r",
  \u015A: "S",
  \u015C: "S",
  \u015E: "S",
  \u0160: "S",
  \u015B: "s",
  \u015D: "s",
  \u015F: "s",
  \u0161: "s",
  \u0162: "T",
  \u0164: "T",
  \u0166: "T",
  \u0163: "t",
  \u0165: "t",
  \u0167: "t",
  \u0168: "U",
  \u016A: "U",
  \u016C: "U",
  \u016E: "U",
  \u0170: "U",
  \u0172: "U",
  \u0169: "u",
  \u016B: "u",
  \u016D: "u",
  \u016F: "u",
  \u0171: "u",
  \u0173: "u",
  \u0174: "W",
  \u0175: "w",
  \u0176: "Y",
  \u0177: "y",
  \u0178: "Y",
  \u0179: "Z",
  \u017B: "Z",
  \u017D: "Z",
  \u017A: "z",
  \u017C: "z",
  \u017E: "z",
  \u0132: "IJ",
  \u0133: "ij",
  \u0152: "Oe",
  \u0153: "oe",
  \u0149: "'n",
  \u017F: "s"
}, Ee = Le(Fe), Me = Ee, Pe = Me, Be = s, We = /[\xc0-\xd6\xd8-\xf6\xf8-\xff\u0100-\u017f]/g, ze = "\\u0300-\\u036f", De = "\\ufe20-\\ufe2f", Ne = "\\u20d0-\\u20ff", Ze = ze + De + Ne, Ge = "[" + Ze + "]", He = RegExp(Ge, "g");
function Ve(o) {
  return o = Be(o), o && o.replace(We, Pe).replace(He, "");
}
var qe = Ve, Je = /[^\x00-\x2f\x3a-\x40\x5b-\x60\x7b-\x7f]+/g;
function Ke(o) {
  return o.match(Je) || [];
}
var Ye = Ke, Qe = /[a-z][A-Z]|[A-Z]{2}[a-z]|[0-9][a-zA-Z]|[a-zA-Z][0-9]|[^a-zA-Z0-9 ]/;
function Xe(o) {
  return Qe.test(o);
}
var or = Xe, _ = "\\ud800-\\udfff", er = "\\u0300-\\u036f", rr = "\\ufe20-\\ufe2f", nr = "\\u20d0-\\u20ff", tr = er + rr + nr, L = "\\u2700-\\u27bf", F = "a-z\\xdf-\\xf6\\xf8-\\xff", ur = "\\xac\\xb1\\xd7\\xf7", cr = "\\x00-\\x2f\\x3a-\\x40\\x5b-\\x60\\x7b-\\xbf", ar = "\\u2000-\\u206f", ir = " \\t\\x0b\\f\\xa0\\ufeff\\n\\r\\u2028\\u2029\\u1680\\u180e\\u2000\\u2001\\u2002\\u2003\\u2004\\u2005\\u2006\\u2007\\u2008\\u2009\\u200a\\u202f\\u205f\\u3000", E = "A-Z\\xc0-\\xd6\\xd8-\\xde", sr = "\\ufe0e\\ufe0f", M = ur + cr + ar + ir, P = "['\u2019]", I = "[" + M + "]", pr = "[" + tr + "]", B = "\\d+", mr = "[" + L + "]", W = "[" + F + "]", z = "[^" + _ + M + B + L + F + E + "]", fr = "\\ud83c[\\udffb-\\udfff]", dr = "(?:" + pr + "|" + fr + ")", vr = "[^" + _ + "]", D = "(?:\\ud83c[\\udde6-\\uddff]){2}", N = "[\\ud800-\\udbff][\\udc00-\\udfff]", c = "[" + E + "]", lr = "\\u200d", $ = "(?:" + W + "|" + z + ")", br = "(?:" + c + "|" + z + ")", S = "(?:" + P + "(?:d|ll|m|re|s|t|ve))?", C = "(?:" + P + "(?:D|LL|M|RE|S|T|VE))?", Z = dr + "?", G = "[" + sr + "]?", gr = "(?:" + lr + "(?:" + [vr, D, N].join("|") + ")" + G + Z + ")*", xr = "\\d*(?:1st|2nd|3rd|(?![123])\\dth)(?=\\b|[A-Z_])", Ir = "\\d*(?:1ST|2ND|3RD|(?![123])\\dTH)(?=\\b|[a-z_])", $r = G + Z + gr, Sr = "(?:" + [mr, D, N].join("|") + ")" + $r, Cr = RegExp([
  c + "?" + W + "+" + S + "(?=" + [I, c, "$"].join("|") + ")",
  br + "+" + C + "(?=" + [I, c + $, "$"].join("|") + ")",
  c + "?" + $ + "+" + S,
  c + "+" + C,
  Ir,
  xr,
  B,
  Sr
].join("|"), "g");
function hr(o) {
  return o.match(Cr) || [];
}
var yr = hr, Rr = Ye, Tr = or, Ar = s, Or = yr;
function jr(o, e, r) {
  return o = Ar(o), e = r ? void 0 : e, e === void 0 ? Tr(o) ? Or(o) : Rr(o) : o.match(e) || [];
}
var Ur = jr, kr = ke, wr = qe, _r = Ur, Lr = "['\u2019]", Fr = RegExp(Lr, "g");
function Er(o) {
  return function(e) {
    return kr(_r(wr(e).replace(Fr, "")), o, "");
  };
}
var Mr = Er, Pr = je, Br = Mr, Wr = Br(function(o, e, r) {
  return e = e.toLowerCase(), o + (r ? Pr(e) : e);
}), zr = Wr;
const Nr = {
  install(o, e = {}) {
    const r = { componentPrefix: "", ...e };
    Object.entries({
      Select: /* @__PURE__ */ Object.assign({ "./components/Select/Custom.vue": () => import("./Custom.d2579f1d.js"), "./components/Select/Standard.vue": () => import("./Standard.da16ea9a.js") }),
      Form: /* @__PURE__ */ Object.assign({ "./components/Form/Button.vue": () => import("./Button.335b90ca.js"), "./components/Form/Checkbox.vue": () => import("./Checkbox.f787d8ee.js"), "./components/Form/InputWrapper.vue": () => import("./InputWrapper.6ce7c17b.js"), "./components/Form/Password.vue": () => import("./Password.bc181375.js"), "./components/Form/Select.vue": () => import("./Select.be197f82.js"), "./components/Form/Separator.vue": () => import("./Separator.41c80dfe.js"), "./components/Form/TextArea.vue": () => import("./TextArea.be0a0328.js"), "./components/Form/TextInput.vue": () => import("./TextInput.88239f53.js"), "./components/Form/Title.vue": () => import("./Title.80b94e26.js") }),
      Block: /* @__PURE__ */ Object.assign({ "./components/Block/Burger.vue": () => import("./Burger.07f9d73d.js"), "./components/Block/Flash.vue": () => import("./Flash.f294be6f.js"), "./components/Block/FrontVersion.vue": () => import("./FrontVersion.47c10dcf.js"), "./components/Block/StarRating.vue": () => import("./StarRating.1ff079b4.js") }),
      Icon: /* @__PURE__ */ Object.assign({ "./components/Icon/ArrowLeft.vue": () => import("./ArrowLeft.3fca1dfc.js"), "./components/Icon/ArrowRight.vue": () => import("./ArrowRight.b933bf0f.js"), "./components/Icon/ArrowRightUpFromSquare.vue": () => import("./ArrowRightUpFromSquare.b48cb1a7.js"), "./components/Icon/ArrowToTop.vue": () => import("./ArrowToTop.a424bc01.js"), "./components/Icon/ArrowsRotate.vue": () => import("./ArrowsRotate.08223820.js"), "./components/Icon/Baby.vue": () => import("./Baby.af6a9733.js"), "./components/Icon/BadgePercent.vue": () => import("./BadgePercent.b84339a7.js"), "./components/Icon/Bars.vue": () => import("./Bars.38007052.js"), "./components/Icon/BookOpen.vue": () => import("./BookOpen.c4fe4375.js"), "./components/Icon/Books.vue": () => import("./Books.9b1bb9e2.js"), "./components/Icon/BrandFacebookSquare.vue": () => import("./BrandFacebookSquare.b9f8956a.js"), "./components/Icon/BrandInstagram.vue": () => import("./BrandInstagram.346fff27.js"), "./components/Icon/BrandPinterestSquare.vue": () => import("./BrandPinterestSquare.d1498a8e.js"), "./components/Icon/BrandTwitterSquare.vue": () => import("./BrandTwitterSquare.fbe6284a.js"), "./components/Icon/Bullhorn.vue": () => import("./Bullhorn.38cfb723.js"), "./components/Icon/CalendarCheck.vue": () => import("./CalendarCheck.37b07557.js"), "./components/Icon/CalendarCircleUser.vue": () => import("./CalendarCircleUser.33670d6d.js"), "./components/Icon/CalendarClock.vue": () => import("./CalendarClock.70f0d75d.js"), "./components/Icon/CalendarXMark.vue": () => import("./CalendarXMark.01656231.js"), "./components/Icon/ChartLine.vue": () => import("./ChartLine.6da360a8.js"), "./components/Icon/Check.vue": () => import("./Check.19baa068.js"), "./components/Icon/ChevronDown.vue": () => import("./ChevronDown.13e9ebdb.js"), "./components/Icon/ChevronLeft.vue": () => import("./ChevronLeft.3227788e.js"), "./components/Icon/ChevronRight.vue": () => import("./ChevronRight.5f947231.js"), "./components/Icon/Child.vue": () => import("./Child.c2137f00.js"), "./components/Icon/CircleCheck.vue": () => import("./CircleCheck.7694e892.js"), "./components/Icon/CircleInfo.vue": () => import("./CircleInfo.41068772.js"), "./components/Icon/CirclePlus.vue": () => import("./CirclePlus.580a074b.js"), "./components/Icon/ClipBoardList.vue": () => import("./ClipBoardList.c4fbc764.js"), "./components/Icon/Clock.vue": () => import("./Clock.9ba3b199.js"), "./components/Icon/Close.vue": () => import("./Close.949273d7.js"), "./components/Icon/CodeScanner.vue": () => import("./CodeScanner.33881e99.js"), "./components/Icon/Coins.vue": () => import("./Coins.c9a6ada3.js"), "./components/Icon/CreditCard.vue": () => import("./CreditCard.d3a54cfd.js"), "./components/Icon/Delete.vue": () => import("./Delete.15c83286.js"), "./components/Icon/DogLeashed.vue": () => import("./DogLeashed.52194f4a.js"), "./components/Icon/Download.vue": () => import("./Download.49bf9665.js"), "./components/Icon/Edit.vue": () => import("./Edit.b3ec9c1e.js"), "./components/Icon/EllipsisVertical.vue": () => import("./EllipsisVertical.c0d4ceff.js"), "./components/Icon/Envelope.vue": () => import("./Envelope.7cff3220.js"), "./components/Icon/Eye.vue": () => import("./Eye.d2986177.js"), "./components/Icon/File.vue": () => import("./File.1f3b3cdb.js"), "./components/Icon/Hashtag.vue": () => import("./Hashtag.61299b01.js"), "./components/Icon/House.vue": () => import("./House.086831a8.js"), "./components/Icon/Image.vue": () => import("./Image.9f6fd085.js"), "./components/Icon/LocationArrow.vue": () => import("./LocationArrow.9adb176c.js"), "./components/Icon/LocationDot.vue": () => import("./LocationDot.e527eee7.js"), "./components/Icon/Lock.vue": () => import("./Lock.55b06039.js"), "./components/Icon/Logout.vue": () => import("./Logout.6046ef21.js"), "./components/Icon/MoneyBillWay.vue": () => import("./MoneyBillWay.3824fb62.js"), "./components/Icon/PaperPlane.vue": () => import("./PaperPlane.201e0eaf.js"), "./components/Icon/Person.vue": () => import("./Person.e941c3bd.js"), "./components/Icon/PhotoFilm.vue": () => import("./PhotoFilm.9878f31a.js"), "./components/Icon/Podium.vue": () => import("./Podium.50fab7e1.js"), "./components/Icon/Question.vue": () => import("./Question.57099800.js"), "./components/Icon/Rectangle.vue": () => import("./Rectangle.60cd8db5.js"), "./components/Icon/Search.vue": () => import("./Search.c296e93d.js"), "./components/Icon/Send.vue": () => import("./Send.59fd132b.js"), "./components/Icon/Share.vue": () => import("./Share.3af4008d.js"), "./components/Icon/ShareSquare.vue": () => import("./ShareSquare.b8e3832e.js"), "./components/Icon/Star.vue": () => import("./Star.32168e1b.js"), "./components/Icon/Tag.vue": () => import("./Tag.e0c27449.js"), "./components/Icon/ThumbsDown.vue": () => import("./ThumbsDown.72cd35f9.js"), "./components/Icon/ThumbsUp.vue": () => import("./ThumbsUp.2cfb8190.js"), "./components/Icon/Trash.vue": () => import("./Trash.89f4905d.js"), "./components/Icon/Upload.vue": () => import("./Upload.5c2429b8.js"), "./components/Icon/User.vue": () => import("./User.127bf588.js"), "./components/Icon/UserCog.vue": () => import("./UserCog.614a63f6.js"), "./components/Icon/UserGroup.vue": () => import("./UserGroup.5963555b.js"), "./components/Icon/UserLock.vue": () => import("./UserLock.e425056d.js"), "./components/Icon/Verified.vue": () => import("./Verified.0f24a35a.js"), "./components/Icon/Wallet.vue": () => import("./Wallet.177084a4.js"), "./components/Icon/Webinar.vue": () => import("./Webinar.a8b2bfe4.js") })
    }).forEach(([n, u]) => {
      Object.entries(u).forEach(([H, V]) => {
        const q = w(
          zr(H.split("/").pop().replace(/\.\w+$/, ""))
        );
        o.component(`${r.componentPrefix}${n}${q}`, J(V));
      });
    }), o.config.globalProperties.$formKitConfigs = r, o.provide("formKitConfigs", r);
  }
};
export {
  Nr as default
};
